File: python/sublist/sublist.py
SUBLIST = 'sub'
SUPERLIST = 'sup'
EQUAL = 'eq'
UNEQUAL = 'uneq'

def check_lists(first, second):
    if first == second:
        return EQUAL
    elif _contains(first, second):
        return SUBLIST
    elif _contains(second, first):
        return SUPERLIST
    return UNEQUAL

def _contains(needle, haystack):
	span = len(needle)
	r = len(haystack) - span + 1
	return any(True for idx in range(r) if haystack[idx:span+idx] == needle)

File: javascript/sublist/sublist.js
class Sublist {

    constructor(list = []) {
        this.list = list;
    }

    compare(list2) {
        if (equal(this.list, list2.list)) {
            return 'EQUAL';
        }
        if (contains(this.list, list2.list)) {
            return 'SUBLIST';
        }
        if (contains(list2.list, this.list)) {
            return 'SUPERLIST';
        }
        return 'UNEQUAL';
    }
}

const contains = (needle, haystack) => {
    return haystack.some((_, idx) => {
        const sub = haystack.slice(idx, idx + needle.length);
        return equal(needle, sub);
    });
}

const equal = (l1, l2) => {
    return l1.length === l2.length
        && l1.every((val, idx) => val === l2[idx]);
}

module.exports = Sublist;